govcd/vdc_group.go

```diff
@@ -0,0 +1,478 @@
+	if err != nil {
+		return nil, err
+	}
+
+	var wrappedVdcGroups []*VdcGroup
+	for _, response := range responses {
+		urlRef, err := adminOrg.client.OpenApiBuildEndpoint(endpoint, response.Id)
+		if err != nil {
+			return nil, err
+		}
+		wrappedVdcGroup := &VdcGroup{
+			VdcGroup: response,
+			client:   adminOrg.client,
+			Href:     urlRef.String(),
+			parent:   adminOrg,
+		}
+		wrappedVdcGroups = append(wrappedVdcGroups, wrappedVdcGroup)
+	}
+
+	return wrappedVdcGroups, nil
+}
+
+// GetVdcGroupByName retrieves VDC group by given name
+// When the name contains commas, semicolons or asterisks, the encoding is rejected by the API in VCD 10.2 version.
+// For this reason, when one or more commas, semicolons or asterisks are present we run the search brute force,
+// by fetching all VDC groups and comparing the names. Yet, this not needed anymore in VCD 10.3 version.
+// Also, url.QueryEscape as well as url.Values.Encode() both encode the space as a + character. So we use
+// search brute force too. Reference to issue:
+// https://github.com/golang/go/issues/4013
+// https://github.com/czos/goamz/pull/11/files
+func (adminOrg *AdminOrg) GetVdcGroupByName(name string) (*VdcGroup, error) {
+	slowSearch, params, err := isShouldDoSlowSearch("name", name, adminOrg.client)
+	if err != nil {
+		return nil, err
+	}
+
+	vdcGroups, err := adminOrg.GetAllVdcGroups(params)
+	if err != nil {
+		return nil, err
+	}
+	if len(vdcGroups) == 0 {
+		return nil, ErrorEntityNotFound
+	}
+
+	if slowSearch {
+		for _, vdcGroup := range vdcGroups {
+			if vdcGroup.VdcGroup.Name == name {
+				return vdcGroup, nil
+			}
+		}
+		return nil, ErrorEntityNotFound
+	}
+
+	if len(vdcGroups) > 1 {
+		return nil, fmt.Errorf("more than one VDC group found with name '%s'", name)
+	}
+	return vdcGroups[0], nil
+}
+
+// GetVdcGroupById Returns VDC group using provided ID
+func (adminOrg *AdminOrg) GetVdcGroupById(id string) (*VdcGroup, error) {
+	tenantContext, err := adminOrg.getTenantContext()
+	if err != nil {
+		return nil, err
+	}
+
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcGroups
+	minimumApiVersion, err := adminOrg.client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if id == "" {
+		return nil, fmt.Errorf("empty VDC group ID")
+	}
+
+	urlRef, err := adminOrg.client.OpenApiBuildEndpoint(endpoint, id)
+	if err != nil {
+		return nil, err
+	}
+
+	vdcGroup := &VdcGroup{
+		VdcGroup: &types.VdcGroup{},
+		client:   adminOrg.client,
+		Href:     urlRef.String(),
+		parent:   adminOrg,
+	}
+
+	err = adminOrg.client.OpenApiGetItem(minimumApiVersion, urlRef, nil, vdcGroup.VdcGroup, getTenantContextHeader(tenantContext))
+	if err != nil {
+		return nil, err
+	}
+
+	return vdcGroup, nil
+}
+
+// Update updates existing Vdc group. Allows changing only name and description and participating VCDs
```

Comment:
A cross reference between functions Update and GenericUpdate would be nice to navigate better

Reply:
added